packages/experimental/TabList/src/TabListAnimatedIndicator/useAnimatedIndicatorStyles.ts

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import type { ViewStyle } from 'react-native';
+
+import type { AnimatedIndicatorStyles } from './TabListAnimatedIndicator.types';
+import type { TabLayoutInfo } from '../TabList/TabList.types';
+import { TabListContext } from '../TabList/TabListContext';
+
+/**
+ * This hook handles logic for generating the styles for the TabList's Animated Indicator.
+ */
+export function useAnimatedIndicatorStyles(): AnimatedIndicatorStyles {
+  const { animatedIndicatorStyles, layout, selectedKey, vertical } = React.useContext(TabListContext);
+
+  const selectedIndicatorLayout = React.useMemo<TabLayoutInfo | null>(() => {
+    return selectedKey ? layout.tabs[selectedKey] : null;
+  }, [selectedKey, layout.tabs]);
+
+  // Calculate styles using both layout information and user defined styles
+  const styles = React.useMemo<AnimatedIndicatorStyles | null>(() => {
+    // if not all layout props have been recorded for the current selected indicator, don't render the animated indicator
+    if (selectedIndicatorLayout) {
```

Comment:
Code will be a little less nested if you instead put the "bail" condition at the top of the function. Ie

if (!selectedIndicatorLayout) {
    return null;
}

... rest of function


Imo it also makes the function easier to read since it will list all the reasons for exiting early first.